stm32_part_table.yaml

```diff
@@ -181,7 +181,7 @@ stm32f4:
       - STM32F401
 
   stm32f405:
-    url: https://www.st.com/en/microcontrollers/stm32f405-microprocessors-415.html
+    url: https://www.st.com/en/microcontrollers/stm32f405-415.html
```

Comment:
Following this link redirects to `https://www.st.com/en/microcontrollers-microprocessors/stm32f405-415.html` so perhaps we should use that instead?